src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -275,6 +275,21 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             Analyze(compilation, method, node, diagnostics, useMethodSignatureReturnType: false, useMethodSignatureParameterTypes: false, methodSignatureOpt: null, returnTypes: null, initialState: null, callbackOpt);
         }
 
+        internal static void AnalyzeIfNeeded(
+            CSharpCompilation compilation,
+            BoundAttribute attribute,
+            DiagnosticBag diagnostics)
+        {
+            if (compilation.LanguageVersion < MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion() ||
+                attribute.Constructor is null ||
```

Comment:
>attribute.Constructor is null [](start = 16, length = 29)

Do we have nothing to analyze in this case? Not even the arguments and property/field initializers?

Reply:
In theory we could analyze the field/property assignment, but everything in NullableWalker is done in the context of a method body, so it's difficult. We could probably pick a default constructor, then not do any analysis on the ctor arguments in that case? Not sure how much that buys us though...

---
In reply to: [264457880](https://github.com/dotnet/roslyn/pull/33816#discussion_r264457880) [](ancestors = 264457880)

Reply:
Fixed the walker so we can analyze without the method symbol. Added tests to check that we do so correctly.

---
In reply to: [264853179](https://github.com/dotnet/roslyn/pull/33816#discussion_r264853179) [](ancestors = 264853179,264457880)